### PR TITLE
Smmp batch not divisible by microbatches fix

### PR DESCRIPTION
**Pull request by mansimane**
# What does this PR do?

Fixes: 
Batch size not divisible by microbatches issue in sagemaker model parallel. Following is summary of changes:
1. Updated SequentialDistributedSampler to generate samples of multiples of batchsize.
2. Updated preds_gatherer and labels_gatherer calls to be multiple of batch size.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@sgugger @philschmid @anirudh2290

### CODE REVIEW

File: src/transformers/trainer.py [hunk 1]
```diff
@@ -1783,8 +1783,8 @@ def prediction_loop(
 
         eval_losses_gatherer = DistributedTensorGatherer(world_size, num_examples, make_multiple_of=batch_size)
         if not prediction_loss_only:
-            preds_gatherer = DistributedTensorGatherer(world_size, num_examples)
-            labels_gatherer = DistributedTensorGatherer(world_size, num_examples)
+            preds_gatherer = DistributedTensorGatherer(world_size, num_examples,  make_multiple_of=batch_size)
+            labels_gatherer = DistributedTensorGatherer(world_size, num_examples,  make_multiple_of=batch_size)
```
Comment (mansimane) [inline src/transformers/trainer.py:1787]: Make `preds_gatherer` and `labels_gatherer` to be multiple of batchsize

**Comment by sgugger** [comment]
Thanks @mansimane !
I added the test and realized the implementation was not working as expected, so fixed it (I had forgotten this does not behave like the `DistributedSampler` that takes one sample every other num_replicas but sliced the indices at the beginning). If you want to have a last look to check I didn't do anything bad that would be great.

The rest of the changes are just the result of our styling scripts

**Review by philschmid — APPROVED** [review]
LGTM. I tested `run_glue.py` for `mnli` and `mrpc` for microbatch size 4

**Comment by mansimane** [comment]
Changes look good to me too. I test with microbatch size 2.

**Review by LysandreJik — APPROVED** [review]
Great, LGTM!

**Pull request merged**